tests/test_pytypes.cpp

```diff
@@ -13,6 +13,22 @@
 
 #include <utility>
 
+#if defined(PYBIND11_CPP20)
+#    if __has_include(<ranges>) // __has_include has been part of C++17, no need to check it
+#        if !defined(PYBIND11_COMPILER_CLANG)
```

Comment:
It's minor, but could you please try to condense this?
Untested:

```
// llvm/llvm-project#52696
#if !defined(PYBIND11_COMPILER_CLANG) || __clang_major__ >= 16
#define PYBIND11_TEST_PYTYPES_HAS_RANGES
#endif
```

Reply:
I compressed it a bit more.